src/lib/utils/ct_utils.h

```diff
@@ -64,15 +67,89 @@ constexpr inline void unpoison(const T* p, size_t n) {
    BOTAN_UNUSED(p, n);
 }
 
+//
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+//
+// Convenience overloads for poisoning/unpoisoning various types and ranges
+//
+
+template <std::integral T>
+constexpr void poison(T& p) {
+   poison(&p, sizeof(T));
+}
+
+template <std::integral T>
+constexpr void unpoison(T& p) {
```

Comment:
There is earlier in the file a generic `unpoison(T&)` that should probably be removed - I think it only exists precisely for the case of unpoisoning an integer that was somehow derived from a secret.

Reply:
We did remove this; or more precisely: restrict it to `std::integral` here. Otherwise the overloads might have been ambiguous.

In the codebase it was used for integers _and_ CT::Mask unpoisening. The latter is now handled via the new `CT::Mask::const_time_(un)poison` members.

Reply:
Ah, oops, I missed that being deleted already in the diff